### PR TITLE
Migrate wasm languages to dev docs

### PR DESCRIPTION
**Pull request by tpmccallum**
The migration of wasm-languages can now occur in this order, [1177](https://github.com/fermyon/developer/pull/1177/files) -> [939](https://github.com/fermyon/fermyon.com/pull/939) -> [69](https://github.com/fermyon/wasm-languages/pull/69) -> [942](https://github.com/fermyon/fermyon.com/pull/942)

Content must go through a pre-merge checklist.

## Pre-Merge Content Checklist

This documentation has been checked to ensure that:

- [x] The `title`, `template`, and `date` are all set
- [x] Does this PR have a new menu item (anywhere in `templates/*.hbs` files) that points to a document `.md` that is set to publish in the future? If so please only publish the `.md` and `.hbs` changes in real-time (otherwise there will be a menu item pointing to a `.md` file that does not exist)
- [x] File does not use CRLF, but uses plain LF (hint: use `cat -ve <filename> | grep '^M' | wc -l` and expect 0 as a result)
- [x] Has passed [`bart check`](https://developer.fermyon.com/bartholomew/quickstart)

Screenshot 2024-02-20 at 09 09 23

- [x] Has been manually tested by running in Spin/Bartholomew (hint: use `PREVIEW_MODE=1` and run `npm run styles` to update styling)

**Provide "Edit This Page" contribution option:**

Screenshot 2024-02-20 at 10 47 04

**Add template styling for consistency:**

Screenshot 2024-02-20 at 10 46 36

- [x] Headings are using Title Case
- [x] Code blocks have the programming language set to properly highlight syntax and the proper copy directive
- [x] Have tested with `npm run test` and resolved all errors

Screenshot 2024-02-20 at 08 45 53

- [x] Relates to an existing (potentially outdated) blog article? If so please add URL in blog to point to this content.

### CODE REVIEW

File: content/wasm-languages/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+[extra]
+author = "Fermyon Staff"
+type = "page"
+url = "https://github.com/fermyon/developer/blob/main/content/wasm-languages/CONTRIBUTING.md"
+
+---
+
+We welcome contributors. To make changes or additions, create new pull requests against this repository.
+
+The [content template](https://github.com/fermyon/developer/content/wasm-languages/tpl.md) provides the scaffolding for adding a new language. The [About Examples](./about-examples.md) explains how an example should work. (Examples are not required. But when provided, we like to keep them consistent.)
```
[the remark this answers is not in the document]
Comment (itowlson) [reply content/wasm-languages/CONTRIBUTING.md:15]: I would think `tpl.md` would not be served from `developer.fermyon.com` so would need to be specifically the GH URL

**Comment by flynnduism** [comment]
What left side menu will these pages have on the Developer site?

**Comment by tpmccallum** [comment]
Great question @flynnduism 
**Matrix page**
The matrix page (the landing page; entry point) has no left-hand-side menu. It is as wide as possible to allow the table/matrix to display. 
**Language pages**
The matrix page has links to each of the language pages.
All language pages have the latest version of Spin in the left-hand-side menu. My thoughts around this were around converting curious first-time Fermyon visitors into clownloads and getting some buy-in.
image

File: content/wasm-languages/webassembly-language-support.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+Spin, Fermyon Platform and Fermyon Cloud require [WASI](https://wasi.dev) support. Any language that has a ✅ for WASI should be supported on the Fermyon Platform. The *Spin SDK* indicates that there is additional libraries available for Spin.
+
+>> If you are interested in contributing to this guide, head on over to [the GitHub repo](https://github.com/fermyon/developer/wasm-languages).
+
+We are often asked which languages are best supported for production-grade WebAssembly. We suggest [C][C]/[C++][CPLUSPLUS], [Rust][Rust], and [AssemblyScript][AssemblyScript].
+
+## Updates and Additions
+
+The source for the WebAssembly Language Guide is located in a [public GitHub project](https://github.com/fermyon/wasm-languages). If you find errors, want to make additions, or have further corrections for us, the [issue queue](https://github.com/fermyon/wasm-languages/issues) is a great place to discuss.
```
Comment (tpmccallum) [inline content/wasm-languages/webassembly-language-support.md:102]:
```suggestion
The source for the WebAssembly Language Guide is located in a [public GitHub project](https://github.com/fermyon/developer/wasm-languages). If you find errors, want to make additions, or have further corrections for us, the [issue queue](https://github.com/fermyon/developer/issues) is a great place to discuss.
```

File: content/wasm-languages/webassembly-language-support.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+For each environment, we use the following icons to indicate a level of support:
+
+- ✅  Usable
+- ⏳ In progress
+- ❌ Not implemented
+- `N/A` Not applicable
+
+Spin, Fermyon Platform and Fermyon Cloud require [WASI](https://wasi.dev) support. Any language that has a ✅ for WASI should be supported on the Fermyon Platform. The *Spin SDK* indicates that there is additional libraries available for Spin.
+
+>> If you are interested in contributing to this guide, head on over to [the GitHub repo](https://github.com/fermyon/developer/wasm-languages).
```
Comment (tpmccallum) [inline content/wasm-languages/webassembly-language-support.md:96]:
```suggestion
```

File: content/wasm-languages/tpl.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+<!--
+This section should just talk about how this project is to be used.
+
+For example, TinyGo might show how to compile to WASI
+-->
+
+## Pros and Cons
+
+<!-- List out some pros and cons of this language vs others WHEN IT COMES TO WASM
```
[the remark this answers is not in the document]
Comment (tpmccallum) [reply content/wasm-languages/tpl.md:46]: Not sure why that is all caps lol

File: content/wasm-languages/standards.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+DeisLabs (at Microsoft) proposed Wagi based on the CGI 1.1 specification.
+
+- The architecture document is [in the Wagi server repo](https://github.com/deislabs/wagi/blob/main/docs/architecture.md)
+- The description of the [environment variables](https://github.com/deislabs/wagi/blob/main/docs/environment_variables.md)
+- Spin's [Wagi support](https://developer.fermyon.com/spin/v2/http-trigger)
+
+## Spin Improvement Proposals (SIPs)
+
+- The [source repository](https://github.com/fermyon/spin/tree/main/docs/content/sips)
+- The [published versions](https://spin.fermyon.dev/sips/index)
```
Comment (tpmccallum) [inline content/wasm-languages/standards.md:54]:
```suggestion
```

File: content/wasm-languages/index.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+The [Language Matrix](/wasm-languages/webassembly-language-support) provides an easy reference for which languages support what features of WebAssembly. It also links to detailed pages for each programming language that Fermyon tracks.
+
+Fermyon WebAssembly Languages does not track WebAssembly execution environments or runtimes. For example, we track ways to compile [PHP](/wasm-languages/php) to WebAssembly, but we do not track how to execute WebAssembly in a PHP program.
+
+## Help Us Stay Current
+
+We do our best to keep the information here up to date. But with so many languages, each evolving rapidly, we're bound to miss a thing or two. If you have a correction or update, ping us on Twitter [@fermyontech](https://twitter.com/fermyontech).
+
+>> If you are interested in contributing to this guide, head on over to [the GitHub repo](https://github.com/fermyon/wasm-languages).
```
Comment (tpmccallum) [inline content/wasm-languages/index.md:22]:
```suggestion
>> If you are interested in contributing to this guide, head on over to [the GitHub repo](https://github.com/fermyon/developer/wasm-languages).
```

File: content/wasm-languages/index.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+## Adding Your Language to the Matrix
+
+If you are affiliated with a language project that adds WebAssembly support to a language, ping us on Twitter [@fermyontech](https://twitter.com/fermyontech) and let us know. We are particularly interested in WebAssembly-specific languages as well as WASI support for top-20 languages. When assessing new projects, we do prefer stable and complete codebases over experimental ones. However, given how young the WebAssembly world is, this is not a requirement.
```
Comment (tpmccallum) [inline content/wasm-languages/index.md:26]:
```suggestion
If you are affiliated with a language project that adds WebAssembly support to a language, please consider [contributing](./CONTRIBUTING.md). We are particularly interested in WebAssembly-specific languages and WASI support for top-20 languages. When assessing new projects, we prefer stable and complete codebases over experimental ones. However, given how young the WebAssembly world is, this is not a requirement.
```

File: content/wasm-languages/about-examples.md [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+
+```toml
+[[module]]
+module = "hello.wasm"
+route = "/"
+```
+
+Then we start `wagi` with `wagi -c modules.toml`.
+
+>> If you are interested in contributing to this guide, head on over to [the GitHub repo](https://github.com/fermyon/wasm-languages).
```
Comment (tpmccallum) [inline content/wasm-languages/about-examples.md]:
```suggestion
>> If you are interested in contributing to this guide, head on over to [the GitHub repo](https://github.com/fermyon/developer/wasm-languages).
```

File: content/wasm-languages/README.md [hunk 9]
```diff
@@ -0,0 +1,57 @@
+# Fermyon WebAssembly Language Guide
```
[the remark this answers is not in the document]
Comment (tpmccallum) [reply content/wasm-languages/README.md:1]: It looks like the README is set to not publish on the site. I can strip the README back and perhaps point to where the wasm-languages contribution page is rendered on the web. :heavy_check_mark:

File: content/wasm-languages/webassembly-language-support.md [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+
+For each environment, we use the following icons to indicate a level of support:
+
+- ✅  Usable
+- ⏳ In progress
+- ❌ Not implemented
+- `N/A` Not applicable
+
+Spin, Fermyon Platform and Fermyon Cloud require [WASI](https://wasi.dev) support. Any language that has a ✅ for WASI should be supported on the Fermyon Platform. The *Spin SDK* indicates that there is additional libraries available for Spin.
+
```
Comment (tpmccallum) [inline content/wasm-languages/webassembly-language-support.md:95]:
```suggestion
```
No double lines allowed.

**Review by tpmccallum — COMMENTED** [review]
Remove stray blank line

File: content/wasm-languages/cpp.md [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+
+- The [WebAssembly and C++](https://neugierig.org/software/blog/2022/06/wasm-c++.html) blog post by Evan Martin is a great analysis of the state of C++ dev for WebAssembly.
+- Cheerp is an [LLVM-based toolchain for compiling C++ to Wasm and JavaScript](https://github.com/leaningtech/cheerp-compiler). An overview [blog post](https://medium.com/leaningtech/cheerp-2-7-compile-cpp-to-webassembly-plus-javascript-c9b3ef7e318b) explains the basics.
+- Mozilla has a tutorial on [Compiling C++ to Wasm with Emscripten](https://developer.mozilla.org/en-US/docs/WebAssembly/C_to_wasm)
+- We love this [really cool slide presentation](https://nxxm.github.io/cppcon2018/CPP_EVERYWHERE_WITH_WASM.html#/) with a code walk of C++ in the browser
+- Example of building [C++ WebAssembly files with Clang](https://github.com/PetterS/clang-wasm)
+- The [yo-wasm](https://github.com/deislabs/yo-wasm) tool can help you get started faster. (Yes, we built `yo-wasm`.)
+- The [QT GUI toolkit](https://www.qt.io/) supports [compiling to WebAssembly](https://wiki.qt.io/Qt_for_WebAssembly) for the browser. (We blogged about [LibreOffice in the browser](/blog/qt-libreoffice-wasm). That uses QT.)
```
Comment (tpmccallum) [inline content/wasm-languages/cpp.md:55]:
```suggestion
- The [QT GUI toolkit](https://www.qt.io/) supports [compiling to WebAssembly](https://wiki.qt.io/Qt_for_WebAssembly) for the browser. (We blogged about [LibreOffice in the browser](https://www.fermyon.com/blog/qt-libreoffice-wasm). That uses QT.)
```

File: content/wasm-languages/c-sharp.md [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+tags = ["c#", "csharp", "dotnet", ".net", "webassembly"]
+template = "page_lang"
+[extra]
+author = "Fermyon Staff"
+last_modified = "2022-03-10T21:50:50Z"
+url = "https://github.com/fermyon/developer/blob/main/content/wasm-languages/c-sharp.md"
+
+---
+
+The .NET (aka dotnet) ecosystem supports a variety of languages, including C# (C sharp) and ASP.NET. Programs that target .NET are compiled to an intermediary bytecode format that runs in the .NET Common Language Runtime (CLR). This makes .NET an example of a [virtual machine language](/blog/scripts-vs.compiled-wasm.md).
```
Comment (tpmccallum) [inline content/wasm-languages/c-sharp.md:13]:
```suggestion
The .NET (aka dotnet) ecosystem supports a variety of languages, including C# (C sharp) and ASP.NET. Programs that target .NET are compiled to an intermediary bytecode format that runs in the .NET Common Language Runtime (CLR). This makes .NET an example of a [virtual machine language](https://www.fermyon.com/blog/scripts-vs-compiled-wasm).
```

**Review by tpmccallum — COMMENTED** [review]
Fix a couple of broken links

File: content/wasm-languages/assemblyscript.md [hunk 13]
```diff
@@ -0,0 +1,252 @@
...
+tags = ["assemblyscript", "typescript", "javascript", "webassembly"]
+template = "page_lang"
+[extra]
+author = "Fermyon Staff"
+last_modified = "2022-03-10T21:50:50Z"
+url = "https://github.com/fermyon/developer/blob/main/content/wasm-languages/assemblyscript.md"
+
+---
+
+Inspired by [TypeScript](/wasm-langauges/typescript), AssemblyScript is a strongly typed language. It was written specifically with WebAssembly in mind, and the entire toolchain is oriented around WebAssembly.
```
Comment (tpmccallum) [inline content/wasm-languages/assemblyscript.md:13]:
```suggestion
Inspired by [TypeScript](./typescript.md), AssemblyScript is a strongly typed language. It was written specifically with WebAssembly in mind, and the entire toolchain is oriented around WebAssembly.
```

File: content/wasm-languages/powershell.md [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+tags = ["powershell", "webassembly"]
+template = "page_lang"
+[extra]
+author = "Fermyon Staff"
+url = "https://github.com/fermyon/developer/blob/main/content/wasm-languages/powershell.md"
+
+---
+
+PowerShell is consistently ranked among the top programming languages.
+However, it is not generally considered to be a general purpose language like [C](/wasm-languages/c-lang) or [Python](/wasm-languges/python).
```
Comment (tpmccallum) [inline content/wasm-languages/powershell.md:13]:
```suggestion
However, it is not generally considered to be a general-purpose language like [C](./c-lang.md) or [Python](./python.md).
```

**Review by itowlson — APPROVED** [review]
LGTM once we fix or exclude the red links.  Great work, thanks so much for tackling it!

File: content/wasm-languages/lua.md [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+
+No details. 
+
+## Example
+
+Currently, there are no examples. None of the known implementations have WASI support.
+
+## Learn More
+
+Here are some great resources:
```
Comment (tpmccallum) [inline content/wasm-languages/lua.md:41]:
```suggestion
Resources:
```

File: content/wasm-languages/lua.md [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+
+- This [fun blog post](https://yiwei.dev/port-lua-to-web-environment-using-webassembly/) explains how to tweak Lua to compile the interpreter to WebAssembly.
```
Comment (tpmccallum) [inline content/wasm-languages/lua.md:43]:
```suggestion
```

**Pull request merged**